client/src/components/SettingsDialog.js

```diff
@@ -0,0 +1,290 @@
+                { showPasswordSettings === true
+                 &&
+                 <React.Fragment>
+                    <Grid>
+                        <TextField
+                            label="New password"
+                            fullWidth
+                            type="password"
+                            required
+                            onChange={e => formDispatch({
+                                type: "password",
+                                payload: e.target.value
+                            })}
+                            value={formState.password.value}
+                            error={errorMsg.length !== 0}
+                        />  
+                        <TextField
+                            label="Confirm new password"
+                            fullWidth
+                            type="password"
+                            required
+                            onChange={e => formDispatch({
+                                type: "confirmPassword",
+                                payload: e.target.value
+                            })}
+                            value={formState.confirmPassword.value}
+                            helperText={errorMsg}
+                            error={errorMsg.length !== 0}
+                        />
+                    </Grid>
+                </React.Fragment>
+                }
+                <SettingsHeading>
+                    Edit your profile picture
+                </SettingsHeading>
+                <Grid className="imageContainer">
+                    <img src={image} className="userPic" alt="userPic" />
+                    {/* <Input
+                        name="profileImage"
+                        accept=" .jpg, .png, .jpeg"
+                        className="imageSelect"
+                        type="file"
+                        onChange={(e) => formDispatch({
+                            type: "image",
+                            payload: e.target.files[0], 
+                        })}
+                        error={formState.image.error}
+                    /> */}
+                    {/* <ImageErrorMsg>
+                        {formState.image.error === true && "Please upload a .png, .jpg, or .jpeg file"}
+                    </ImageErrorMsg> */}
+                    <DropZone 
+                        formDispatch={formDispatch}
+                    />
+                </Grid>
+
+                <SettingsHeading>
+                    Change your language.
+                </SettingsHeading>
+                <NativeSelect
+                    value={formState.language}
+                    onChange={(e) => formDispatch({
+                        type: "language",
+                        payload: e.target.value 
+                    })}
+                >
+                    {languages.sort().map(language => (
+                        <option value={language} key={uuidv4()}>{language}</option>
```

Comment:
I like that you thought of adding unique keys to help React render things properly 👍🏻 
Maybe using a library for this is overkill. Keys just need to be unique, so if the language is unique, you can just use that as a key and it will work just fine:

```suggestion
                        <option value={language} key={language}>{language}</option>
```

Reply:
The only caveat to that is to not use the index of the array. Because if things get added or removed to the array, the index of things change and React gets confused as to what is what. Here's a [good article](https://kentcdodds.com/blog/understanding-reacts-key-prop) about understanding the `key` prop in React.